src/commands/start.py

```diff
@@ -236,18 +261,25 @@ async def main() -> None:
     remote_signer_config = None
 
     if len(keystores) == 0:
-        if not settings.remote_signer_url:
-            raise RuntimeError('No keystores and no remote signer URL provided')
-
-        # No keystores loaded but remote signer URL provided
-        remote_signer_config = RemoteSignerConfiguration.from_file(
-            settings.remote_signer_config_file
-        )
-        logger.info(
-            'Using remote signer at %s for %i public keys',
-            settings.remote_signer_url,
-            len(remote_signer_config.pubkeys_to_shares.keys()),
-        )
+        if not settings.remote_signer_url and not settings.hashi_vault_url:
+            raise RuntimeError('No keystores, no remote signer or hashi vault URL provided')
+
+        if settings.hashi_vault_url:
+            # No keystores loaded but hashi vault configuration specified
+            hashi_vault_config = HashiVaultConfiguration.from_settings()
+            logger.info('Using hashi vault at %s for loading public keys')
+            keystores = await load_hashi_vault_keys(hashi_vault_config)
+
+        else:
+            # No keystores loaded but remote signer URL provided
+            remote_signer_config = RemoteSignerConfiguration.from_file(
+                settings.remote_signer_config_file
+            )
+            logger.info(
+                'Using remote signer at %s for %i public keys',
+                settings.remote_signer_url,
+                len(remote_signer_config.pubkeys_to_shares.keys()),
+            )
```

Comment:
```
if settings.hashi_vault_url:
    ...
elif settings.hashi_vault_url:
    ...
else:
    raise RuntimeError('No keystores, no remote signer or hashi vault URL provided')
```

Reply:
Done in https://github.com/stakewise/v3-operator/pull/184/commits/c8ac6ce3ec65cc6a24c66301a0e34f4277447e2c, thanks